nixos/modules/services/networking/nomad.nix

```diff
@@ -49,6 +49,18 @@ in
         '';
       };
 
+      settingsFiles = mkOption {
```

Comment:
I almost feel like it'd be preferable to just allow users to set a `configDir` and let nomad just read all the files in there as configs. Instead of making uses write the config imperatively but then come here and declare the file as valid declaratively.

In other words, if we're going to provide a declarative escape hatch, let's make it nicer to use by not burdening it with declarative boilerplate.

Curious to hear what @aanderse and @nh2 think.

Reply:
Yeah, this could be an option. Is there way to ensure that `configDir` exists without having to put an empty file in it, like the `consul.nix` drv does?

Reply:
I don't think users need write any extra settings files imperatively though. You can generate the extra files however you want. The way I am using this is to generate an extra settings file that has the `datacenter` and `region` configuration values set according to GCE instance metadata pulled using the metadata server. I generate that file by adding `ExecStartPre = "+${someScript}";` to the unit configuration.

Reply:
I just realized that since the `-config` option passed to `nomad agent` accepts directories, that I don't think we need a `configDir` option. @lovesegfault LMK what you think here.

Reply:
I think this is addressed by the fact that you can specify a directory in `extraSettingsPaths`.